23.0.0.2 draft (#2971)
OpenLiberty/blogs: posts/2050-03-07-23.0.0.2.adoc

blog_description: Open Liberty 23.0.0.2 introduces a new capability in Admin Center's Server Config tool to test database connections.  The `server stop` command also receives a new capability to specify the maximum amount of time to wait for the server to stop via the `--timeout` option.  This release also provides many significant bug fixes, including one that addresses a CVE in Jakarta RESTful Web Services 3.0 feature.
open-graph-image: https://openliberty.io/img/blog/ac_db_test_container_auth.png
open-graph-image-alt: Open Liberty Admin Center connection test panel
---
= Test database connections and set server stop time out in Open Liberty 23.0.0.2
Michal Broz <https://github.com/mbroz2>
:imagesdir: /
:url-prefix:
:url-about: /
//Blank line here is necessary before starting the body of the post.

> Open Liberty 23.0.0.2 introduces a new capability in Admin Center's Server Config tool to test database connections.  The `server stop` command also receives a new capability to specify the maximum amount of time to wait for the server to stop via the `--timeout` option.  This release also provides many significant bug fixes, including one that addresses a CVE in Jakarta RESTful Web Services 3.0 feature.

Reviewer: ```suggestion
Open Liberty 23.0.0.2 introduces a new capability to test database connections with the Admin Center Server Config tool.  The `server stop` command adds the the `--timeout` option to specify the maximum amount of time to wait for the server to stop.  This release also provides many significant bug fixes, including one that addresses a CVE in Jakarta RESTful Web Services 3.0 feature.
```

Check out link:{url-prefix}/blog/?search=release&search!=beta[previous Open Liberty GA release blog posts].

[#run]

== Run your apps using 23.0.0.2

If you're using link:{url-prefix}/guides/maven-intro.html[Maven], here are the coordinates:

[source,xml]
----
<dependency>
    <groupId>io.openliberty</groupId>
    <artifactId>openliberty-runtime</artifactId>
    <version>23.0.0.2</version>
    <type>zip</type>
</dependency>
----

Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:

[source,gradle]
----
dependencies {
    libertyRuntime group: 'io.openliberty', name: 'openliberty-runtime', version: '[23.0.0.2,)'
}
----

Or if you're using Docker:

[source]
----
FROM open-liberty
----

Or take a look at our link:{url-prefix}/downloads/[Downloads page].

[link=https://stackoverflow.com/tags/open-liberty]
image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]

// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/24124
// Contact/Reviewer: aknguyen7,ReeceNana
// // // // // // // // 
[#db_test]
== Test database connections using Admin Center   

> Looking for a simple way to test your database connections? With this beta release, you can now validate your connections using the Liberty Admin Center feature. The connection test exercises the same code paths as your applications, giving you confidence in your server configuration. The Admin Center connection validation feature is enabled by REST APIs featured in the link:{url-prefix}/blog/2019/09/13/testing-database-connections-REST-APIs.html[Testing database connections in Open Liberty apps with REST APIs] blog post.

Reviewer: ```suggestion
Looking for a simple way to test your database connections? With this beta release, you can now validate your connections using the Liberty Admin Center feature. The connection test exercises the same code paths as your applications, giving you confidence in your server configuration. The Admin Center connection validation feature is enabled by the same REST APIs featured in the link:{url-prefix}/blog/2019/09/13/testing-database-connections-REST-APIs.html[Testing database connections in Open Liberty apps with REST APIs] blog post.
```

1. Configure a Liberty server using the example `server.xml` file for guidance and then start the server. After the server is started, you can check the logs to find the URL to navigate to the Admin Center. In the previous example, you can navigate to the Admin Center using the `https://localhost:9443/adminCenter/` URL.

2. Select the **Server Config** tool.

+
[.img_border_light]
image::img/blog/ac_db_test_server_config.png[Server Config Tool,width=20%,align="center"]

3. Select **server.xml** to edit.

+
[.img_border_light]
image::img/blog/ac_db_test_serverxml.png[server.xml,width=50%,align="center"]

4. In the **Design > Server** menu, navigate to the resource you want to test and click the **Test** button.

+
[.img_border_light]
image::img/blog/ac_db_test_resource.png[Select resource,width=50%,align="center"]

5. Choose the type of authentication your application uses:

+
* For applications that use container authentication, choose the **Application authentication** tab and select whether to use default authentication, specify an authentication alias, or choose a login module configuration.

+

> For this example, the configuration doesn't specify default authentication on the `dataSource` element or configure any login modules. Therefore, you must specify an authentication alias by using the dropdown field.

Reviewer: ```suggestion
For this example, the configuration doesn't specify default authentication on the `dataSource` element or configure any login modules. Therefore, you must specify an authentication alias by using the drop-down field.
```

+
[.img_border_light]
image::img/blog/ac_db_test_container_auth.png[Container authentication,width=50%,align="center"]

> * For applications that use application authentication, choose the **Application authentication** tab and fill in a valid user name and password for the database resource.

Reviewer: ```suggestion
* For applications that use application authentication, choose the **Application authentication** tab and fill in a valid username and password for the database resource.
```

+
[.img_border_light]
image::img/blog/ac_db_test_app_auth.png[Application authentication,width=50%,align="center"]

> * If your application does not use a resource reference and the server.xml doesn't include `enableContainerAuthForDirectLookups="true"` in the config element, then choose **No resource reference** tab and fill in a valid user name and password for the database resource.

Reviewer: ```suggestion
* If your application does not use a resource reference and the `server.xml` doesn't include `enableContainerAuthForDirectLookups="true"` in the config element, then choose the **No resource reference** tab and fill in a valid username and password for the database resource.
```

6. Click the **Connection Test** button to run the test and display the results.  The following example shows a successful connection test:

[.img_border_light]
image::img/blog/ac_db_test_successful_test.png[Successful connection test example,width=50%,align="center"]

In addition to link:{url-prefix}/docs/latest/reference/feature/jdbc-4.3.html[Java Database Connectivity] you can also test connections to link:{url-prefix}/docs/latest/reference/feature/connectors-2.0.html[Jakarta Connectors], link:{url-prefix}/docs/latest/reference/feature/messaging-3.0.html[Jakarta Messaging] and link:{url-prefix}/docs/latest/reference/feature/cloudant-1.0.html[Cloudant Integration] resources. 

For more information about administering Liberty using a GUI, refer to the link:{url-prefix}/docs/latest/admin-center.html[Manage Open Liberty with Admin Center] documentation.

// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC> 

// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/23282
// Contact/Reviewer: jimblye,ReeceNana
// // // // // // // // 
[#timeout]
== Specify how long to wait for the server to stop

> Open Liberty 23.0.0.2 introduces a `--timeout` command line option for the `server stop` command.  This allows you to specify the maximum amount of time the `server stop` command waits for confirmation that the server has stopped.  This allows you to specify the time out value used when waiting for the server to stop. 

Reviewer: ```suggestion
Open Liberty 23.0.0.2 introduces a `--timeout` command line option for the `server stop` command.  You can use this option to specify the maximum amount of time the `server stop` command waits for confirmation that the server has stopped.  
```

> Prior to this update, there was no way to configure the maximum amount of time of 30 seconds that the server would wait when stopping.

Reviewer: ```suggestion
Prior to this update, the default maximum waiting period of 30 seconds could not be adjusted.
```

> The time out value can be specified in minutes (`m`), seconds (`s`), or a combination of both.  When a unit is not specified, the default of seconds is used.  Minutes and seconds can be combined, for example `2m30s` which means 2 minutes and 30 seconds.  Examples:

Reviewer: ```suggestion
The timeout value can be specified in minutes (`m`), seconds (`s`), or a combination of both.  When a unit is not specified, the default of seconds is used.  Minutes and seconds can be combined, for example `2m30s`, which means 2 minutes and 30 seconds.  Examples:
```

   
[source, xml]
----
   ./server stop                   // 30 seconds
   ./server stop --timeout=45      // 45 seconds
   ./server stop --timeout=45s     // 45 seconds
   ./server stop --timeout=3m20s   // 3 minutes, 20 seconds
----

> The default timeout value is 30 seconds. If the server consistently	takes longer than 30 seconds to stop, consider increasing the timeout value by using the --timeout option.

Reviewer: ```suggestion
The default timeout value is 30 seconds. If the server consistently takes longer than 30 seconds to stop, consider increasing the timeout value by using the --timeout option.
```

   
// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC> 

[#CVEs]
== Security vulnerability (CVE) fixes in this release
[cols="5*"]
|===
|CVE |CVSS Score |Vulnerability Assessment |Versions Affected |Notes

|http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-45787[CVE-2022-45787]
|5.5
|Information disclosure
|21.0.0.12 - 23.0.0.1
|Affects the link:{url-prefix}/docs/latest/reference/feature/restfulWS-3.0.html[feature:restfulWS-3.0] feature
|===

For a list of past security vulnerability fixes, reference the link:{url-prefix}/docs/latest/security-vulnerabilities.html[Security vulnerability (CVE) list].

[#bugs]
== Notable bugs fixed in this release

We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A23002+label%3A%22release+bug%22[full list of bugs fixed in 23.0.0.2].

* link:https://github.com/OpenLiberty/open-liberty/issues/24371[Server fails to start due to conflict on servlet feature]
+

> When individually installing a set of EE7 or EE8 features, the server can fail to start due to a conflict on servlet features.  An example failure is below.

Reviewer: ```suggestion
When individually installing a set of EE7 or EE8 features, the server can fail to start due to a conflict on servlet features, as shown in the following example.
```
